File: frontend/src/types/location.ts
export type LocationSource = 'gps' | 'manual';
export type GPSStatus = 'idle' | 'loading' | 'success' | 'error' | 'denied';

export interface UserLocation {
  latitude: number;
  longitude: number;
  city?: string | null;
  district?: string | null;
  state?: string | null;
  country?: string | null;
  postalCode?: string | null;
  postal_code?: string | null;
  formattedAddress?: string | null;
  formatted_address?: string | null;
  source: LocationSource;
  location_source?: LocationSource;
  displayName?: string;
  display_name?: string;
  updatedAt?: string;
}

export interface CityOption {
  id?: string;
  name: string;
  district: string;
  state: string;
  country: string;
  latitude: number;
  longitude: number;
  pincode?: string;
  display_name?: string;
}

File: backend/app/schemas/location.py
from typing import Optional, List
from pydantic import BaseModel, Field

class LocationResolveRequest(BaseModel):
    latitude: float = Field(..., ge=-90.0, le=90.0)
    longitude: float = Field(..., ge=-180.0, le=180.0)
    source: Optional[str] = "gps"

class NormalizedLocationResponse(BaseModel):
    latitude: float
    longitude: float
    city: Optional[str] = None
    district: Optional[str] = None
    state: Optional[str] = None
    country: Optional[str] = None
    postal_code: Optional[str] = None
    formatted_address: Optional[str] = None
    source: Optional[str] = "gps"
    provider: Optional[str] = None

class LocationResolveResponse(BaseModel):
    success: bool = True
    location: NormalizedLocationResponse

class LocationSearchResult(BaseModel):
    display_name: str
    latitude: float
    longitude: float
    city: Optional[str] = None
    district: Optional[str] = None
    state: Optional[str] = None
    country: Optional[str] = None
    postal_code: Optional[str] = None
    source: Optional[str] = "manual"

class LocationSearchResponse(BaseModel):
    success: bool = True
    results: List[LocationSearchResult]
